android/src/main/java/com/reactnativereadium/reader/BaseReaderFragment.kt

```diff
@@ -38,6 +41,14 @@ abstract class BaseReaderFragment : Fragment() {
     val viewScope = viewLifecycleOwner.lifecycleScope
 
     channel.send(ReaderViewModel.Event.TableOfContentsLoaded(model.publication.tableOfContents))
+    viewScope.launch {
```

Comment:
@alecdhansen why is the `viewScope` being used here?

Reply:
`positions()` and `positionsByReadingOrder()` are suspend functions, so I'm calling them inside a coroutine. By using this scope, everything cancels automatically if the View is destroyed, which should avoid leaks or unnecessary work.

Happy to hear your thoughts if something else would work better!